lib/rspec/support/object_inspector.rb

```diff
@@ -4,19 +4,39 @@ module Support
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
       # @api private
-      def self.inspect(object)
-        if Time === object
+      # rubocop:disable CyclomaticComplexity
+      # rubocop:disable MethodLength
+      def self.inspect(object, nested=false)
+        case object
+        when Time
           format_time(object)
-        elsif defined?(DateTime) && DateTime === object
-          format_date_time(object)
-        elsif defined?(BigDecimal) && BigDecimal === object
-          "#{object.to_s 'F'} (#{object.inspect})"
-        elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
-          object.description
+        when Hash
+          format_hash(object, nested)
         else
-          object.inspect
+          if defined?(DateTime) && DateTime === object
+            format_date_time(object)
+          elsif defined?(BigDecimal) && BigDecimal === object
+            "#{object.to_s 'F'} (#{object.inspect})"
+          elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
+            object.description
+          else
+            object.inspect
+          end
         end
       end
+      # rubocop:enable CyclomaticComplexity
+      # rubocop:enable MethodLength
+
+      # @private
+      def self.format_hash(object, nested)
```

Comment:
@myronmarston I think I understand what your asking for, but if you could just confirm

```
irb(main):008:0> ObjectInspector.inspect({ :a => 5, :b => 6 })
=> "{:a => 5, :b => 6}"
```

When used inside a differ:
```
irb(main):008:0> ObjectInspector.inspect({ :a => 5, :b => 6 })
=> ":a => 5,\n:b => 6,"
```